### ISSUE TITLE
UnderscoresInNumericLiterals flagged for floats

### ISSUE THREAD

**Issue by vanniktech** [arturbosch/detekt#1538]
```kotlin
assertThat(expressionEvaluator.evaluate("π").toDouble()).isEqualTo(3.1415926535897932385)
```

is being flagged by UnderscoresInNumericLiterals and it should not. RC14

**Comment by 3flex** [comment]
Is this issue that it's flagged for floats, or that it's flagging with only a single digit before the decimal point?

I'd expect things like `3000.1234` would be flagged as it would be better as `3_000.1234`, what do you think?

**Comment by arturbosch** [comment]
It is flagged because of x.1415926535897932385 and the rule expects `3.1_415_926_535_897_932_385`.
Do we really want to to allow `3.1415926535897932385`? Yes, phi is known and the number sequence appears familiar for us, but do we want this for other sequences?
No false positive rather open for configuration option here.

**Comment by vanniktech** [comment]
I don't think this rule should work for floats/doubles. Even `7.4_534_454` looks super weird

**Comment by 3flex** [comment]
My strong preference would be only doing this for digits before the period only. Pretty sure there's an IntelliJ intention that works the same way.

Good:
1_000
1_000_000.48272848
0x55_43_F8

Bad:
1_000.345_678_910

**Comment by vanniktech** [comment]
I'd also exclude heg digits. We are used to `0xFFFFFF` and having `0xFF_FF_FF` is super weird.
